jupyter_server_proxy/handlers.py

```diff
@@ -467,13 +521,17 @@ def options(self, path):
         return self.proxy(self.port, path)
 
 
-def setup_handlers(web_app):
+def setup_handlers(web_app, host_whitelist_hook):
     host_pattern = '.*$'
     web_app.add_handlers('.*', [
         (url_path_join(web_app.settings['base_url'], r'/proxy/(\d+)(.*)'),
          LocalProxyHandler, {'absolute_url': False}),
         (url_path_join(web_app.settings['base_url'], r'/proxy/absolute/(\d+)(.*)'),
          LocalProxyHandler, {'absolute_url': True}),
+        (url_path_join(web_app.settings['base_url'], r'/proxy/(.*):(\d+)(.*)'),
+         RemoteProxyHandler, {'absolute_url': False, 'host_whitelist_hook': host_whitelist_hook}),
+        (url_path_join(web_app.settings['base_url'], r'/proxy/absolute/(.*):(\d+)(.*)'),
+         RemoteProxyHandler, {'absolute_url': True, 'host_whitelist_hook': host_whitelist_hook}),
```

Comment:
Do you mean the order here needs changing?

Reply:
Yes, the `RemoteProxyHandler`s need to be added before the `LocalProxyHandler` ones, since Tornado stops at the first match.  If it's an IP, then the `LocalProxyHandler`s match up to the end of the first quad and that one will fire.  More importantly, I think the presence of the ":" is the thing that should make or break Remote vs Local.

Reply:
@rcthomas done!